lib/libze/libze.c

```diff
@@ -2036,6 +2036,52 @@ typedef struct libze_destroy_cbdata {
     libze_destroy_options *options;
 } libze_destroy_cbdata;
 
+
+typedef struct dependents_cbdata {
+    libze_destroy_cbdata *cbd;
+    zfs_handle_t	*cb_target;
+    nvlist_t	*cb_nvl;
+} dependents_cbdata_t;
+
+
+static int
+promote_dependent_datasets(dependents_cbdata_t *cb)
+{
+    nvpair_t *pair;
+    char *name;
+    for (pair = nvlist_next_nvpair(cb->cb_nvl, NULL);
+    pair != NULL;
+    pair = nvlist_next_nvpair(cb->cb_nvl, pair)) {
+        zfs_handle_t *zhp = zfs_open(cb->cbd->lzeh->lzh, nvpair_name(pair),
+                                     ZFS_TYPE_FILESYSTEM);
+        if (zhp != NULL) {
+            int err;
+            name = zfs_get_name(zhp);
+            err = zfs_promote(zhp);
+            zfs_close(zhp);
+            if (err != 0) {
+                return libze_error_set(cb->cbd->lzeh, LIBZE_ERROR_UNKNOWN,
+                                       "Dataset %s could not be promoted\n", name);
+            }
+        }
+    }
+    return (0);
+}
+
+
+static int
+get_dependents(zfs_handle_t *zhp, void *data){
+    dependents_cbdata_t *cbp = data;
+    const char *tname = zfs_get_name(cbp->cb_target);
+    const char *name = zfs_get_name(zhp);
+    if (!(strncmp(tname, name, strlen(tname)) == 0 &&
```

Comment:
whenever you've got a long lived logical statement, this seems to be more readable:

```
/*
 * I'd call strlen() once
 */
ssize_t       tnlen = strlen(tname);
```

Then, that saves you a ton of typing (plus also having the benefit of not calling the same routing over and over...

```
if (!(strncmp(tname, name, tnlen) == 0 && (name[tnlen] == '/' || name[tnlen] == '@'))) {
    /*
     * body of conditional
     */
}
```

or if that goes beyond the 80 column limit...

```
if (!(strncmp(tname, name, tnlen) == 0 && \
    (name[tnlen] == '/'                || \
    name[tnlen] == '@'))) {
        /*
         * Body of conditional
         */
}
```